viewing_party/party.py

```diff
@@ -1,23 +1,149 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if title == None or genre == None or rating == None:
+        return None
+
+    new_movie = {"title": title, "genre": genre, "rating": rating}
+    return new_movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    for i in range(len(user_data["watchlist"])):
+        if user_data["watchlist"][i]["title"] == title:
+            movie_watched = user_data["watchlist"][i]
+            del user_data["watchlist"][i]
+            user_data["watched"].append(movie_watched)  
+    return user_data

```

Comment:
In the future, try to avoid modifying a list as you're looping over it. You still pass the tests here, but it can lead to some strange bugs! It would be best to modify the list after the loop is finished.